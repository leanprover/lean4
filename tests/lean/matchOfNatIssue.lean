example (x : Int) (h : x = 2) : Int.div 2 1 = x := by
  simp [Int.div]
  trace_state
  simp [h]

example (n : Nat) : Int.div (Int.ofNat n) (Int.ofNat 0) = Int.ofNat (n / 0) := by
  simp [Int.div]

example (n : Nat) : Int.div (Int.ofNat n) 0 = Int.ofNat (n / 0) := by
  simp [Int.div]

example (n : Nat) : Int.mul (Int.ofNat n) (Int.ofNat 0) = Int.ofNat (n * 0) := by
  simp [Int.mul]

example (n : Nat) : Int.mul (Int.ofNat n) 0 = Int.ofNat (n * 0) := by
  simp [Int.mul]
