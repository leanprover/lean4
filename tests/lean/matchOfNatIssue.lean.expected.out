x : Int
h : x = 2
‚ä¢ Int.ofNat (2 / 1) = x
